doc/scheduled-configurations/scheduled-configurations-hld.md

```diff
@@ -0,0 +1,583 @@
+      - [Config Time Range](#config-time-range)
+      - [Show Time Range](#show-time-range)
+      - [Show Scheduled Configurations](#show-scheduled-configurations)
+  - [SAI API](#sai-api)
+  - [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+  - [Restrictions/Limitations](#restrictionslimitations)
+  - [Testing Requirements/Design](#testing-requirementsdesign)
+    - [Unit Test cases](#unit-test-cases)
+    - [System Test cases](#system-test-cases)
+  - [Open/Action items](#openaction-items)
+  - [References](#references)
+
+## Revision
+
+Revision 1.3
+
+## Scope
+
+The scope of this document is to outline the High-Level Design (HLD) for the *Scheduled Configurations* feature in SONiC. This feature aims to set SONiC configurations based on a schedule defined by the user. It includes the ability to schedule configurations for absolute-time events as well as recurring time intervals.
+
+## Definitions/Abbreviations
+
+| **Term**    | **Definition**                                                                       |
+| ----------- | ------------------------------------------------------------------------------------ |
+| *CONFIG_DB* | Configuration Database used by SONiC to store the static configuration of the system |
+| *STATE_DB*  | State Database in SONiC to store the runtime state of the system                     |
+| *SWSS*      | SONiC SWitch State service                                                           |
+| *cron*      | time-based job scheduler linux application                                           |
+
+## Overview
+
+The *Scheduled Configurations* feature will allow administrators to schedule specific network configurations to be applied and later removed or reconfigured at predefined times without manual intervention. This capability is particularly useful for network scenarios where policies need to change dynamically at certain times of the day or week, such as different ACL rules that should be active only during business hours. The scheduled configuration feature easily integrates within the existing infrastructure of SONiC by following established practices and utilizing widely adopted tools like `cron`. `cron` is a time-based job scheduler in Unix-like operating systems, used to automate the execution of scripts and commands at specified times. This approach not only minimizes the learning curve associated with the adoption of new features but also leverages the proven reliability and efficiency of an open-source utility.
+
+## Requirements
+
+### Design Requirements
+
+- **Configurability**: Administrators must be able to define time range intervals that will activate and deactivate configurations. These configurations can update the CONFIG DB during activation and deactivation.
+- **Generic**: The design should be able to configure all existing configurations available in CONFIG DB,  such as ACL, QoS, port shutdown, etc...
```

Comment:
If the changes are limited to these tables, we may consider extended exist table fields to accept time-varying variable (TIME_RANGE entry) as value:
```
    "PORT": {
        "Ethernet1": {
            "lanes": "25,26,27,28",
            "alias": "etp7",
            "index": "1",
            "speed": "10000",
            // "admin_status": "up"/"down",
            "admin_status_variable: "MaintenanceWindow",
            "mtu": "9100"
        }
    },
    "TIME_RANGE": {
        "MaintenanceWindow": {
            "start": "30 6 25 DEC *",
            "end": "0 8 2 JAN *",
            "start_year": "2024",
            "end_year": "2025",
            "start_value": "up",
            "end_value": "down"
        }
    }
```